sonar-ruby-plugin/src/main/java/org/sonarsource/ruby/plugin/SimpleCovSensor.java

```diff
@@ -141,12 +144,39 @@ private static void mergeHitPerLines(Map<String, Map<Integer, Integer>> coverage
     for (int i = 0; i < hitsPerLine.size(); i++) {
       Object hits = hitsPerLine.get(i);
       // Hits can be a Long (coverage data available), null or "ignored".
-      if (hits instanceof Long) {
+      if (hits == null || hits instanceof Long) {
         int line = i + 1;
         Integer currentHits = fileCoverage.getOrDefault(line, 0);
-        fileCoverage.put(line, currentHits + ((Long) hits).intValue());
+        Integer mergedHits = hits == null ?
+          mergeHitsForLine(null, currentHits) :
+          mergeHitsForLine(((Long) hits).intValue(), currentHits);
+        fileCoverage.put(line, mergedHits);
+
+      }
```

Comment:
What about making this code more explicit by splitting the if with two conditions into an it-else-if?
```
      int line = i + 1;
      Integer currentHits = fileCoverage.getOrDefault(line, 0);
      // Hits can be a Long (coverage data available), null or "ignored".
      if (hits instanceof Long) {
        fileCoverage.put(line,  mergeHitsForLine(((Long) hits).intValue(), currentHits));
      }
      else if(hits == null) {
        fileCoverage.put(line, mergeHitsForLine(null, currentHits));
      }
```